tutorials/serverless-grafana-with-iap/index.md

```diff
@@ -0,0 +1,156 @@
+
+You should have a Google Cloud Platform account and project setup, billing configured for your project, and Terraform installed and enabled.
+  1. [Create or select a Google Cloud project.](https://console.cloud.google.com/project)
+  1. [Enable billing for your project.](https://support.google.com/cloud/answer/6293499#enable-billing)
+  1. Choose a region to host your project in, ideally one that’s close to you. You can find an overview of available regions [here](https://cloud.google.com/compute/docs/regions-zones).
+  1. Make sure you know the domain name where you want to host your Grafana dashboard and are able to edit the A record for this domain.
+
+
+### Configure the OAuth consent screen
+
+Configure an OAuth consent screen for Identity-Aware Proxy. 
+  1. Go to GCP Console > Security > Identity-Aware Proxy. 
+  2. If you didn’t configure a consent screen before, there will be a red warning message prompting you to configure one. 
+  ![Consent missing error message](./iap-consent-not-configured.png)
+  Depending on your setup you can either enable the consent screen for internal or external users. Internal allows only users that are part of your Cloud Identity organisation to use Grafana, but requires you to have a Cloud Identity organisation. You can alternativley select external, which allows you to also add users that are outside your organisation. As long as your app wasn't publish, you will need to whitelist "testers" that can access your application (see optional step 4).
+  3. Click Configure Consent Screen, choose User Type Internal or External depending on your needs. Add a name and support email addresses and click Create.
+  ![Configure consent](./iap-configure-oauth.png)
+  4. (Optional, only applies if your selected external before) In this step you can add users as test users to your application. You will need to also grant those users the role "IAP-Secured Web App User " which we will do at a later stage.
+  ![Add test users](./iap-consent-test-users.png)
+  5. Enter the app name and user support email, then click **Save** and continue until the process is complete.
+
+### Set up your Environment
+
+Next, you’re going to set up the environment in order for the project to deploy.
+  1. [Open a new Cloud Shell session.](https://console.cloud.google.com/?cloudshell=true)
+  1. *Run* `git clone https://github.com/GoogleCloudPlatform/community.git` to download the sources to your cloud shell.
+  1. `cd ./community/tutorials/serverless-grafana-with-iap/code` change directory to the *code* folder.
+  1. *Set* required environment variables. Replace [YOUR_DOMAIN] with the domain on which you want to host your Grafana dashboard.
+  ```
+  export TF_VAR_project_id=$GOOGLE_CLOUD_PROJECT
+  export TF_VAR_domain=[YOUR_DOMAIN]
+  ```
+  1. *Run* `terraform init`
+
+
+### Execute the Terraform script to create your Grafana Dashboard
+
+  1. *Run* `terraform plan` and confirm that all steps are correct.
+  2. *Run* `terraform apply`.
```

Comment:
I received this error when trying to apply:

google_iap_client.project_client: Creating...
╷
│ Error: Error creating Client: googleapi: Error 400: Precondition check failed.
│
│   with google_iap_client.project_client,
│   on lb.tf line 79, in resource "google_iap_client" "project_client":
│   79: resource "google_iap_client" "project_client" {

Reply:
Seems like you can only create "internal" apps (as defined in the OAuth consent screen) with Terraform. You get this error with "external" apps.
https://github.com/hashicorp/terraform-provider-google/issues/8843

It works when I changed to